### PR TITLE
feat: Cascader.Panel support

### PR DESCRIPTION
**Pull request by zombieJ**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |       Cascader support Cascader.Panel component for inline usage.    |
| 🇨🇳 Chinese |     Cascader 支持 Cascader.Panel 组件供内联使用。    |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 3a65190</samp>

This pull request adds a new `CascaderPanel` component that can be used as a standalone cascader panel for inline view cases. It also refactors the cascader style logic and the checkable logic by using custom hooks and extracting reusable functions. It updates the documentation and the dependency for the cascader component.

### <samp>🤖 Generated by Copilot at 3a65190</samp>

*  Add a new `CascaderPanel` component that provides a standalone cascader panel for inline view cases ([link](https://github.com/ant-design/ant-design/pull/45089/files?diff=unified&w=0#diff-52e1ddb7e1c662939e80e99f72b99eaf0b8da20296c837a6e6cc9fcc09a48d58R1-R55), [link](https://github.com/ant-design/ant-design/pull/45089/files?diff=unified&w=0#diff-4737f6ef564486c449ec019f2826176d8b8d186088e5e912ad06ad1919b2d804L32-R36), [link](https://github.com/ant-design/ant-design/pull/45089/files?diff=unified&w=0#diff-4737f6ef564486c449ec019f2826176d8b8d186088e5e912ad06ad1919b2d804R351), [link](https://github.com/ant-design/ant-design/pull/45089/files?diff=unified&w=0#diff-4737f6ef564486c449ec019f2826176d8b8d186088e5e912ad06ad1919b2d804R364))
*  Add documentation and code examples for the `CascaderPanel` component in both Chinese and English ([link](https://github.com/ant-design/ant-design/pull/45089/files?diff=unified&w=0#diff-80ccc5da2583c6f9d9999a4beaeed94ef63db2d0abd8344ff8653ecb3cf7af4aR1-R7), [link](https://github.com/ant-design/ant-design/pull/45089/files?diff=unified&w=0#diff-126fe8be94340e44631f8221f455ba751fc69e623656756cc0e6a6fad5aa3ffeR1-R57), [link](https://github.com/ant-design/ant-design/pull/45089/files?diff=unified&w=0#diff-5ced7e6e1259091e502d7b0512332fdeb05319823a37c5ae3fa7ffb1ccf8aa50R39), [link](https://github.com/ant-design/ant-design/pull/45089/files?diff=unified&w=0#diff-64c67e20c18f62d937b9af3bc101a2a1b9cd92df3ac46d93a8e252683297b5f3R40))
*  Extract the logic of rendering the checkbox element based on the `multiple` prop to a custom hook `useCheckable` and reuse it in both `Cascader` and `CascaderPanel` components ([link](https://github.com/ant-design/ant-design/pull/45089/files?diff=unified&w=0#diff-fbcef4451b8a1f679970429199c46f01d1aa025ae415c92200da3002a8adcc8bR1-R8), [link](https://github.com/ant-design/ant-design/pull/45089/files?diff=unified&w=0#diff-4737f6ef564486c449ec019f2826176d8b8d186088e5e912ad06ad1919b2d804L273-R275))
*  Extract the logic of generating the CSS style for the cascader columns based on the component token to a function `getColumnsStyle` and reuse it in both `Cascader` and `CascaderPanel` components ([link](https://github.com/ant-design/ant-design/pull/45089/files?diff=unified&w=0#diff-9d96d7e729614395a22c393efc9b29ee933f5b16d67972bab03f5c99898a9ceeR1-R119), [link](https://github.com/ant-design/ant-design/pull/45089/files?diff=unified&w=0#diff-ea9836bf0b7334709965e0995571f43a0466a5b7d66d2e5093b611e706c8f07fL2-R7), [link](https://github.com/ant-design/ant-design/pull/45089/files?diff=unified&w=0#diff-ea9836bf0b7334709965e0995571f43a0466a5b7d66d2e5093b611e706c8f07fL78-R72), [link](https://github.com/ant-design/ant-design/pull/45089/files?diff=unified&w=0#diff-8a22481590f9bcba223f0f520b4dbe3b39d1e425b089aceb25c79ba64a3b7371R1-R41))
*  Export the function `prepareComponentToken` that prepares the component token for the cascader style logic and reuse it in the panel style logic ([link](https://github.com/ant-design/ant-design/pull/45089/files?diff=unified&w=0#diff-ea9836bf0b7334709965e0995571f43a0466a5b7d66d2e5093b611e706c8f07fL190-R110), [link](https://github.com/ant-design/ant-design/pull/45089/files?diff=unified&w=0#diff-8a22481590f9bcba223f0f520b4dbe3b39d1e425b089aceb25c79ba64a3b7371R1-R41))
*  Remove the redundant style logic for the checkbox element and the unused variables from the `Cascader` style logic ([link](https://github.com/ant-design/ant-design/pull/45089/files?diff=unified&w=0#diff-ea9836bf0b7334709965e0995571f43a0466a5b7d66d2e5093b611e706c8f07fL46-R51), [link](https://github.com/ant-design/ant-design/pull/45089/files?diff=unified&w=0#diff-ea9836bf0b7334709965e0995571f43a0466a5b7d66d2e5093b611e706c8f07fL71-L72))
*  Update the version of the `rc-cascader` dependency that supports the `Panel` component ([link](https://github.com/ant-design/ant-design/pull/45089/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L126-R126))

### CODE REVIEW

File: components/cascader/Panel.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  const { getPrefixCls, renderEmpty, direction: rootDirection } = React.useContext(ConfigContext);
+
+  const mergedDirection = direction || rootDirection;
+  const cascaderPrefixCls = getPrefixCls('cascader', customizePrefixCls);
+
+  const [, hashId] = useStyle(cascaderPrefixCls);
+  usePanelStyle(cascaderPrefixCls);
+
+  // ===================== Empty =====================
+  const mergedNotFoundContent = notFoundContent || renderEmpty?.('Cascader') || (
```
Comment (kiner-tang) [inline components/cascader/Panel.tsx:36]: 这样如果传 null 或者是 false 是不是没办法隐藏 NotFoundContent

File: components/cascader/demo/panel.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+## zh-CN
+
+`5.10.0` 新增，适用于一些需要内嵌适用的场景。
```
Comment (kiner-tang) [inline components/cascader/demo/panel.md:3]: 不是可以在后面的 `<code version="5.10.0">` 增加版本角标吗？这里是不是不需要额外加？

File: components/cascader/index.en-US.md [hunk 3]
```diff
@@ -36,6 +36,7 @@ Cascade selection box.
 <code src="./demo/custom-dropdown.tsx">Custom dropdown</code>
 <code src="./demo/placement.tsx">Placement</code>
 <code src="./demo/status.tsx">Status</code>
+<code src="./demo/panel.tsx">Panel</code>
```
Comment (kiner-tang) [inline components/cascader/index.en-US.md:39]:
```suggestion
<code src="./demo/panel.tsx" version="5.10.0">Panel</code>
```

File: components/cascader/index.zh-CN.md [hunk 4]
```diff
@@ -37,6 +37,7 @@ demo:
 <code src="./demo/custom-dropdown.tsx">扩展菜单</code>
 <code src="./demo/placement.tsx">弹出位置</code>
 <code src="./demo/status.tsx">自定义状态</code>
+<code src="./demo/panel.tsx">面板使用</code>
```
Comment (kiner-tang) [inline components/cascader/index.zh-CN.md:40]:
```suggestion
<code src="./demo/panel.tsx" version="5.10.0">面板使用</code>
```

File: components/cascader/Panel.tsx (hunk 1 above)
Comment (zombieJ) [reply components/cascader/Panel.tsx:36]: 内联空必须要要有个占位，否则会留一个空的框，会很怪异。

File: components/cascader/demo/panel.md (hunk 2 above)
Comment (zombieJ) [reply components/cascader/demo/panel.md:3]: 666，我都忘记这个了。

File: components/cascader/Panel.tsx (hunk 1 above)
Comment (kiner-tang) [reply components/cascader/Panel.tsx:36]: 但个人感觉，使用者如果明确传了一个 null 或者是 false 过来，就是明确不想要这个东西，而是想用其他方式提示，既然类型上支持 null 和 false 的话，是不是不应该打破这个规则呢

Comment (zombieJ) [reply components/cascader/Panel.tsx:36]: false 有需求的话可以支持一下，但是这一版我还是想先坚持一下不做这个处理。

**Review by MadCcc — APPROVED** [review]

**Pull request merged**